src/dataframe/dataframe.jl

```diff
@@ -914,7 +916,7 @@ end
 hcat!(df::DataFrame, x::AbstractVector; makeunique::Bool=false) =
     hcat!(df, DataFrame(AbstractVector[x]), makeunique=makeunique)
 hcat!(x::AbstractVector, df::DataFrame; makeunique::Bool=false) =
```

Comment:
This method doesn't make sense to me. How about removing it?

Reply:
We could leave `hcat` to accept only `AbstractDataFrames` I think. Prepending or appending a vector to a `DataFrame` is problematic anyway - as the column name is autogenerated.

Do we want to deprecate both?

Reply:
I hadn't realized `hcat!` was only internal. I'm less concerned with `hcat(::AbstractVector, ::AbstractDataFrame)` (generating names is consistent with what we in the constructor from matrices). As you prefer.